parsing/dialects/bigquery/build.gradle.kts

```diff
@@ -33,3 +36,47 @@ dependencies {
     testImplementation(project(":core", "testClasses"))
     testImplementation(project(":parsing", "testClasses"))
 }
+
+val dialectGenerate by tasks.registering(org.apache.calcite.buildtools.parser.DialectGenerateTask::class) {
+  rootDirectory.set(file("$rootDir/parsing"))
+  dialectDirectory.set(file("."))
+  outputFile = "build/generated/templates/parserImpls.ftl"
+}
+
+// TODO remove these two tasks, temporary until generation task is finished
+tasks.register("test1") {
+    doLast {
+        mkdir("build/generated/templates")
+    }
+}
+tasks.register<Copy>("test2") {
+    from("$rootDir/parsing/parserImpls.ftl")
+    into("$buildDir/generated/templates")
+}
+
+val fmppMain by tasks.registering(org.apache.calcite.buildtools.fmpp.FmppTask::class) {
+    dependsOn(dialectGenerate)
+    dependsOn("test1") // TODO remove
+    dependsOn("test2") // TODO remove
+
+    inputs.dir(".")
+    config.set(file("config.fmpp"))
+    templates.set(file("$rootDir/parsing/src/main/resources"))
+}
+
+val javaCCMain by tasks.registering(org.apache.calcite.buildtools.javacc.JavaCCTask::class) {
+    dependsOn(fmppMain)
+    lookAhead.set(2)
+    val parserFile = fmppMain.map {
+        it.output.asFileTree.matching { include("**/Parser.jj") }.singleFile
+    }
+    inputFile.set(parserFile)
+    packageName.set("org.apache.calcite.sql.parser.bigquery")
+}
+
+ide {
+    fun generatedSource(javacc: TaskProvider<org.apache.calcite.buildtools.javacc.JavaCCTask>, sourceSet: String) =
```

Comment:
nit: line looks a little long, can you move sourceSet to a second line?

Reply:
Yeah I'll change that in the other gradle files too